### ISSUE TITLE
Prevent Engineer & Administrator access to Mod Platform from non-MoJ Devices

### ISSUE THREAD

**Issue by mikereiddigital** [ministryofjustice/modernisation-platform#7536]
### User Story

As a Modernisation Platform team member
I want Administrator and Engineer role access to platform accounts blocked when not using MOJ supplied devices.
So that we meet the NSCS recommendations to implement technical controls to ensure that AWS accounts are only accessed from devices with a known good security posture.

This will be achieved by blocking the access to the AdministratorAccess and ModernisationPlatformEngineer roles unless the connection's outgoing IP address is from the PaloAlto's GlobalProtect gateway using a Deny condition and the "NotIPAddress" rule as shown in this example - https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_aws_deny-ip.html

There are two parts to this work:

1. Adding the required IP & fqdn rules to GlobalProtect so that these requests are sent through the PA gateway, so allowing the known outgoing addresses to be applied. These rules will include addresses (and wildcards) for all of the AWS endpoints that we in Mod Platform are likely to use - so including cli, sdk and api access across the required regions e.g. *.eu-west-2.amazonaws.com.

2. Once this is done and the rules are in place and confirmed as working, add a DENY condition to the policies for the AdministratorAccess & ModernisationPlatformEngineer roles as described above. 

This should be tested in Sprinkler first before applying elsewhere.

Note - this excludes SuperAdmins.

### Value / Purpose

This is required to demonstrate that the platform is implementing key NCSC recommendations to ensure access is only obtained from devices with a known good security posture.

### Useful Contacts

Harvey, Ian <Ian.Harvey1@justice.gov.uk> Carden, Martin <Martin.Carden@justice.gov.uk>

### Additional Information

The outgoing IPs of the PaloAlto infrastructure (which hosts GlobalProtect) will be used the addresses for the condition. This is managed within Technology Services and the contacts are listed above. Ian Harvey is the Senior Technical Architect and Martin Carden is the Senior Infrastructure Engineer who is responsible for the rules.

### Definition of Done

- [ ] Confirm & agree the full list of endpoints we will be adding as rules.
- [ ] Work with the Technology Services teams & the contacts provided to add the required required new GlobalProtect rules & confirm as working. Note that Technology Services require a change request & CAB to apply this rules and there could be a delay whilst this happens. The TS team will provide the outgoing IPs for the GP & PA infrastructure.
- [ ] Apply the required deny conditions to the policies for AdministratorAccess and ModernisationPlatformEngineer roles in Sprinkler using the outgoing IPs provided by the TS team and test. Only proceed to the next stage if overall agreement is reached that the testing works.
- [ ] Confirm with team and apply the deny first to the ModernisationPlatformEngineer role & test. Once confirmed as working, then apply to AdministratorAccess.
- [ ] Amend User Guide docs that cover MP team access to the platform.

Note - before commencing with the change, ensure that the SuperAdmins account is accessible as adding a DENY condition will block access to these two roles if applied incorrectly.

**Comment by mikereiddigital** [comment]
List of AWS addresses & endpoints in scope:

- moj.awsapps.com (MOJ aws access portal)
- *.aws.amazon.com (covers general console access)
- *.amazonaws.com (covers all regional and global endpoints)
-*.cloudfront.net (cloudfront access)

**Comment by mikereiddigital** [comment]
Had a discussion earlier with the team on the scope of the roles to be impacted and what issues we would need to take into account. This diagram was discussed:


![Image](https://github.com/user-attachments/assets/81b142fd-86a6-44bf-b9e4-1db58092304a)

The following was agreed:
- That only the ModernisationPlatformEngineer role would be in scope for now & that this will be tested when the GlobalProtect filter changes are applied.
- SuperAdmin access would be tested to confirm access.
- AdministratorAccess is out of scope for now given that it will be monitored for use in the future.
- Some of the MP team will be testing Prisma Access and possible issues associated with that.

Until we've had confirmation that the GP changes are applied, this ticket will be moved back to the to-do.

**Comment by mikereiddigital** [comment]
Martin Carden has raised the change request - CHG0083668. He mentioned that HMCTS require 10 days notice for these changes so there will be a wait.
